ibis/expr/types/groupby.py

```diff
@@ -62,9 +66,18 @@ def __init__(
         self, table, by, having=None, order_by=None, window=None, **expressions
     ):
         self.table = table
-        self.by = [_get_group_by_key(table, v) for v in util.promote_list(by)] + [
-            _get_group_by_key(table, v).name(k) for k, v in expressions.items()
-        ]
+        self.by = list(
```

Comment:
Can we add a reusable ` group_key_from()` rule?

Reply:
Then converting GroupedTable to annotable should be easier.

Reply:
Fine in a follow-up too.

Reply:
Let's do it in a follow-up.